### PR TITLE
Guard triggering of default watch go routine with enablement flag.

### PR DESCRIPTION
**Pull request by bianpengyuan**
fix https://github.com/istio/istio/issues/36541

### CODE REVIEW

**Review by Monkeyanator — APPROVED** [review]

File: pilot/pkg/leaderelection/leaderelection.go [hunk 1]
```diff
@@ -75,7 +75,9 @@ type LeaderElection struct {
 
 // Run will start leader election, calling all runFns when we become the leader.
 func (l *LeaderElection) Run(stop <-chan struct{}) {
-	go l.defaultWatcher.Run(stop)
+	if l.prioritized {
+		go l.defaultWatcher.Run(stop)
```
Comment (howardjohn) [inline pilot/pkg/leaderelection/leaderelection.go:79]: very nit: should we check `l.defaultWatcher != nil`? This is for sure correct, but a change somewhere else could make the logic slightly off in the future?

**Review by howardjohn — APPROVED** [review]
cc @Monkeyanator

File: pilot/pkg/leaderelection/leaderelection.go (hunk 1 above)
Comment (bianpengyuan) [reply pilot/pkg/leaderelection/leaderelection.go:79]: Updated to also check on l.defaultWatcher is not nil.

**Pull request merged**